### PR TITLE
Add jax.scipy.special.lambertw (k=0 and k=-1, real and complex)

### PR DESCRIPTION
**Pull request by ahmedtaha100**
Implement the Lambert W function for the principal branch (k=0) and the secondary real branch (k=-1), supporting both real and complex inputs.

- Piecewise initial guesses with dual-start Halley for complex k=0 to ensure principal-branch convergence
- 20-iteration Halley refinement via lax.fori_loop
- custom_jvp with derivative 1/(exp(W)*(1+W))
- Explicit SciPy-matching handling for complex infinities, NaN passthrough, and branch-point special values

Addresses jax-ml/jax#13680

### CODE REVIEW

**Comment by ahmedtaha100** [comment]
@jakevdp @hawkinsp

**Comment by ahmedtaha100** [comment]
@gemini-code-assist review

**Comment by ahmedtaha100** [comment]
@gemini review

File: jax/_src/scipy/special.py [hunk 1]
```diff
@@ -2668,6 +2668,302 @@ def _poch_m_derivative(z, m):
...
+  cubic convergence rate (typically converges in 3-5 iterations; the
+  extra headroom handles the k=-1 branch near the branch point where
+  convergence can be slower).
+  """
+  def body(i, w):
+    # Use scaled residual f/exp(w) = w - z*exp(-w) to avoid exp(w)
+    # overflow for large w (float32 overflows at w > 88) and underflow
+    # for k=-1 where w -> -inf.
+    exp_mw = jnp.exp(-w)
+    f_scaled = w - z * exp_mw
```
[the remark this answers is not in the document]
Comment (ahmedtaha100) [reply jax/_src/scipy/special.py:2757]: This is a real issue, but the Halley iteration already converges before w gets that extreme. I'll wait for a human reviewer before taking action.

**Comment by jakevdp** [comment]
Thanks for the contribution! This implementation is pretty complicated, and I'm not sure the JAX team can sign up to be responsible for maintaining it. What would you think about releasing it in a separate package that others can import if needed?

Some context: we've accepted contributions like this in the past, and then down the line it turns out the functions have poor performance and/or accuracy in some domains, and there's no good way to fix it. So we're stuck with a semi-deprecated, half-working API that we are unable to fix, and that we can't remove because enough downstream users depend on it. I don't want to repeat that mistake here.
